prov/efa/src/rdm/efa_rdm_pke_utils.c

```diff
@@ -61,34 +62,36 @@ ssize_t efa_rdm_pke_init_payload_from_ope(struct efa_rdm_pke *pke,
 	assert(tx_iov_index < ope->iov_count);
 	assert(tx_iov_offset < ope->iov[tx_iov_index].iov_len);
 	iov_mr = ope->desc[tx_iov_index];
-
+	p2p_available = false;
+	inline_send = false;
 	/* When using EFA device, EFA device can access memory that
 	 * that has been registered, and p2p is allowed to be used.
 	 */
 	if (iov_mr) {
 		ret = efa_rdm_ep_use_p2p(pke->ep, iov_mr);
 		if (ret < 0)
 			return ret;
-		efa_can_access_memory = ret;
+		p2p_available = ret;
 	/**
 	 * efa can directly access memory if the buffer is host memory
 	 * and the pkt size (hdr + data) is within inline buf size.
 	 */
 	} else if (!efa_mr_is_hmem(iov_mr) &&
 	           payload_offset + data_size <= efa_rdm_ep_domain(pke->ep)->device->efa_attr.inline_buf_size) {
-		efa_can_access_memory = true;
-	} else {
-		efa_can_access_memory = false;
+		inline_send = true;
 	}
 
 	/*
 	 * Copy can be avoid if the following 2 conditions are true:
-	 * 1. EFA can directly access the memory
-	 * 2. data to be send is in 1 iov, because device only support 2 iov, and we use
-	 *    1st iov for header.
+	 * 1. data to be send is in 1 iov, because device only support 2 iov,
+	 * and we use 1st iov for header.
+	 * 2. EFA can directly access the memory
+	 * Note that FI_INJECT requires outbound buffer to be reusable
+	 * immediately after function returns, so without the inline HW support,
```

Comment:
Can you rephrase the `without the inline HW support,` to `unless using inline`

Reply:
Also can we change `EFA can directly access the memory` to `p2p is available or using inline` here